### PR TITLE
Feature/gazeable alert

### PR DESCRIPTION
**Pull request by stevefosterwta**
Added alerts that are presented in Settings warning a user if they will lose head tracking. 
Alerts are presented for:
- Toggling on/off head tracking
- Tapping on privacy and contact developer

### CODE REVIEW

File: EyeSpeak/Features/Settings/SettingsCollectionViewController.swift [hunk 1]
```diff
@@ -147,28 +147,24 @@ class SettingsCollectionViewController: UICollectionViewController, MFMailCompos
         let item = dataSource.snapshot().itemIdentifiers[indexPath.item]
         switch item {
         case .headTrackingToggle:
-            AppConfig.isHeadTrackingEnabled.toggle()
+            if AppConfig.isHeadTrackingEnabled {
+                let alertViewController = GazeableAlertViewController.make { AppConfig.isHeadTrackingEnabled.toggle() }
+                present(alertViewController, animated: true)
+                alertViewController.setAlertTitle("Turn off head tracking? Really?")
```
Comment (stevefosterwta) [inline EyeSpeak/Features/Settings/SettingsCollectionViewController.swift:153]: This is not the final copy.

File: EyeSpeak/Features/Settings/SettingsCollectionViewController.swift [hunk 2]
```diff
@@ -147,28 +147,24 @@ class SettingsCollectionViewController: UICollectionViewController, MFMailCompos
...
+                AppConfig.isHeadTrackingEnabled.toggle()
+            }
+
         case .privacyPolicy:
-            let url = URL(string: "https://vocable.app/privacy.html")!
-            UIApplication.shared.open(url, options: [:], completionHandler: nil)
-            
+            let alertViewController = GazeableAlertViewController.make { self.presentPrivacyAlert() }
+            present(alertViewController, animated: true)
+            alertViewController.setAlertTitle("You’re about to be taken outside of the app. You may lose head tracking control.")
```
Comment (stevefosterwta) [inline EyeSpeak/Features/Settings/SettingsCollectionViewController.swift:161]: Same here. Not the final copy.

File: EyeSpeak/Features/Settings/SettingsCollectionViewController.swift [hunk 3]
```diff
@@ -147,28 +147,24 @@ class SettingsCollectionViewController: UICollectionViewController, MFMailCompos
...
-            let composeVC = MFMailComposeViewController()
-            composeVC.mailComposeDelegate = self
-            composeVC.setToRecipients(["vocable@willowtreeapps.com"])
-            composeVC.setSubject("Feedback for Vocable v\(versionAndBuildNumber)")
-            self.composeVC = composeVC
-            
-            self.present(composeVC, animated: true, completion: nil)
+            let alertViewController = GazeableAlertViewController.make { self.presentEmail() }
+            present(alertViewController, animated: true)
+            alertViewController.setAlertTitle("You’re about to be taken outside of the app. You may lose head tracking control.")
```
Comment (stevefosterwta) [inline EyeSpeak/Features/Settings/SettingsCollectionViewController.swift:166]: Same as above.

File: EyeSpeak/Features/Settings/SettingsCollectionViewController.swift [hunk 4]
```diff
@@ -201,11 +197,38 @@ class SettingsCollectionViewController: UICollectionViewController, MFMailCompos
...
+
+    private func presentPrivacyAlert() {
+        let url = URL(string: "https://vocable.app/privacy.html")!
+        UIApplication.shared.open(url, options: [:], completionHandler: nil)
+    }
+
+    private func presentEmail() {
+
+        guard MFMailComposeViewController.canSendMail() else {
+            NSLog("Mail composer failed to send mail", [])
```
Comment (pg8wood) [inline EyeSpeak/Features/Settings/SettingsCollectionViewController.swift:211]: Can we add a gaze-able version of the alert that was removed in #73?

**Review by pg8wood — APPROVED** [review]
Looks awesome. I'd probably hit up Kayce about the final copy before we merge just so we don't accidentally send it out there

File: EyeSpeak/Common/Views/GazeableButton.swift [hunk 5]
```diff
@@ -54,34 +59,34 @@ class GazeableButton: UIButton {
     }
     
     func sharedInit() {
```
Comment (Clstroud) [inline EyeSpeak/Common/Views/GazeableButton.swift]: Can this be privatized?

**Review by Clstroud — APPROVED** [review]
Looks good, just one small comment 👍

File: EyeSpeak/Features/Settings/SettingsCollectionViewController.swift (hunk 4 above)
Comment (stevefosterwta) [reply EyeSpeak/Features/Settings/SettingsCollectionViewController.swift:211]: `GazeableAlertViewController` has `cancel` and `confirm` buttons so we'd need to refactor to allow a single `Ok` button.

Comment (stevefosterwta) [reply EyeSpeak/Features/Settings/SettingsCollectionViewController.swift:211]: I'm going to do this refactoring is a separate PR.

**Pull request merged**